### ISSUE TITLE
"helm lint --strict" fails for charts created with "helm create" on Helm 2

### ISSUE THREAD

**Issue by jmlrt** [helm/helm#7923]
Output of `helm version`: 
```
Client: &version.Version{SemVer:"v2.16.5", GitCommit:"89bd14c1541fa93a09492010030fd3699ca65a97", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.5", GitCommit:"89bd14c1541fa93a09492010030fd3699ca65a97", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.1", GitCommit:"7879fc12a63337efff607952a323df90cdc7a335", GitTreeState:"clean", BuildDate:"2020-04-10T21:53:58Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): docker-desktop

When running `helm lint --strict` just after creating a chart with `helm create` command, this is failing with `executing "nginx-2.serviceAccountName" at <.Values.serviceAccount.name>: map has no entry for key "name"` error.

Running the same test with Helm 3 is successful.

Helm charts generated by `helm create` command should be compliant by default to `helm lint --strict`.

```shell
➜  helm git:(master) ✗ /usr/local/Cellar/helm@2/2.16.5/bin/helm version     
Client: &version.Version{SemVer:"v2.16.5", GitCommit:"89bd14c1541fa93a09492010030fd3699ca65a97", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.5", GitCommit:"89bd14c1541fa93a09492010030fd3699ca65a97", GitTreeState:"clean"}
➜  helm git:(master) ✗ /usr/local/Cellar/helm@2/2.16.5/bin/helm create nginx-2
Creating nginx-2
➜  helm git:(master) ✗ /usr/local/Cellar/helm@2/2.16.5/bin/helm lint nginx-2  
==> Linting nginx-2
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures
➜  helm git:(master) ✗ /usr/local/Cellar/helm@2/2.16.5/bin/helm lint --strict nginx-2
==> Linting nginx-2
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "nginx-2/templates/serviceaccount.yaml": template: nginx-2/templates/_helpers.tpl:52:53: executing "nginx-2.serviceAccountName" at <.Values.serviceAccount.name>: map has no entry for key "name"

Error: 1 chart(s) linted, 1 chart(s) failed
➜  helm git:(master) ✗ 
➜  helm git:(master) ✗ 
➜  helm git:(master) ✗ /usr/local/Cellar/helm/3.1.2_1/bin/helm version               
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
➜  helm git:(master) ✗ /usr/local/Cellar/helm/3.1.2_1/bin/helm create nginx-3
Creating nginx-3
➜  helm git:(master) ✗ /usr/local/Cellar/helm/3.1.2_1/bin/helm lint nginx-3  
==> Linting nginx-3
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
➜  helm git:(master) ✗ /usr/local/Cellar/helm/3.1.2_1/bin/helm lint --strict nginx-3
==> Linting nginx-3
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
➜  helm git:(master) ✗ k version                                                                                                           
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.1", GitCommit:"7879fc12a63337efff607952a323df90cdc7a335", GitTreeState:"clean", BuildDate:"2020-04-10T21:53:58Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

Related to https://github.com/helm/helm/issues/3822#issuecomment-573772324

**Comment by technosophos** [comment]
I just successfully reproduced on `dev-v2` following the steps above.

**Comment by technosophos** [comment]
Marking closed with #7926
